SpatialGDK/Source/Private/Interop/SnapshotManager.cpp

```diff
@@ -0,0 +1,191 @@
+// It does this by having an entity query for all entities that are not the GSM (workaround for not having the ability to make a query for all entities).
+// Once it has the response to this query, it will send deletion requests for all found entities and then one for the GSM itself.
+// Should only be triggered by the worker which is authoritative over the GSM.
+void USnapshotManager::WorldWipe(const USpatialNetDriver::ServerTravelDelegate& Delegate)
+{
+	UE_LOG(LogSnapshotManager, Warning, TEXT("World wipe for deployment has been triggered. All entities will be deleted."));
+
+	Worker_Constraint GSMConstraint;
+	GSMConstraint.constraint_type = WORKER_CONSTRAINT_TYPE_ENTITY_ID;
+	GSMConstraint.entity_id_constraint.entity_id = GlobalStateManager->GlobalStateManagerEntityId;
+
+	Worker_NotConstraint NotGSMConstraint;
+	NotGSMConstraint.constraint = &GSMConstraint;
+
+	Worker_Constraint WorldConstraint;
+	WorldConstraint.constraint_type = WORKER_CONSTRAINT_TYPE_NOT;
+	WorldConstraint.not_constraint = NotGSMConstraint;
+
+	Worker_EntityQuery WorldQuery{};
+	WorldQuery.constraint = WorldConstraint;
+	WorldQuery.result_type = WORKER_RESULT_TYPE_SNAPSHOT;
+
+	Worker_RequestId RequestID;
+	RequestID = NetDriver->Connection->SendEntityQueryRequest(&WorldQuery);
+
+	EntityQueryDelegate WorldQueryDelegate;
+	WorldQueryDelegate.BindLambda([this, Delegate](Worker_EntityQueryResponseOp& Op)
+	{
+		if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
+		{
+			UE_LOG(LogSnapshotManager, Error, TEXT("SnapshotManager WorldWipe - World entity query failed: %s"), UTF8_TO_TCHAR(Op.message));
+		}
+
+		if (Op.result_count == 0)
+		{
+			UE_LOG(LogSnapshotManager, Error, TEXT("SnapshotManager WorldWipe - No entities found in world entity query"));
+		}
+
+		if (Op.result_count >= 1)
+		{
+			// Send deletion requests for all entities found in the world entity query.
+			DeleteEntities(Op);
+
+			// Also make sure that we kill the GSM.
+			NetDriver->Connection->SendDeleteEntityRequest(GlobalStateManager->GlobalStateManagerEntityId);
+		}
+
+		Delegate.ExecuteIfBound();
+	});
+
+	Receiver->AddEntityQueryDelegate(RequestID, WorldQueryDelegate);
+}
+
+void USnapshotManager::DeleteEntities(const Worker_EntityQueryResponseOp& Op)
+{
+	UE_LOG(LogSnapshotManager, Warning, TEXT("Deleting %u entities."), Op.result_count);
+
+	for (uint32_t i = 0; i < Op.result_count; i++)
+	{
+		UE_LOG(LogSnapshotManager, Log, TEXT("Sending delete request for: %i"), Op.results[i].entity_id);
+		NetDriver->Connection->SendDeleteEntityRequest(Op.results[i].entity_id);
+	}
+}
+
+// GetSnapshotPath will take a snapshot (without the .snapshot extension) name and convert it to a relative path in the Game/Content folder.
+FString GetSnapshotPath(const FString& SnapshotName)
+{
+	FString SnapshotsDirectory = FPaths::ProjectContentDir().Append("Spatial/Snapshots/");
+	return SnapshotsDirectory + SnapshotName + ".snapshot";
```

Comment:
Could someone `accidentally add `.snapshot`? If they do could we not append `.snapshot`?

Reply:
Handled.